### PR TITLE
Update base_head.py

### PR DESCRIPTION
**Pull request by xinxinxinxu**
A small modification that applies to different backbone outputs. from issue #1838

## Motivation

when I use mmpose 1.0.0 rc to train a new backbone, I find that the output head raises a size unmatching error, which did not happen in the past editions. The main issue is the BaseHead treat output of backbone as a tuple by default, while the output of the backbone i used is a tensor.  The positioning of the reported error is ambiguous, which may result in more users consuming some time to locate what the problem is. However, it is only a matter of condition statements.

## Modification

Add a condition statements in a private function of BaseHead class (._transform_inputs) :
original:
image
modified
image

## BC-breaking (Optional)

No

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by Ben-Louis** [comment]
Hi @xinxinxinxu, thanks for your contribution to MMPose! Could you please sign the CLA? It is a necessary step for merging this PR.

**Comment by xinxinxinxu** [comment]
> Hi @xinxinxinxu, thanks for your contribution to MMPose! Could you please sign the CLA? It is a necessary step for merging this PR.

Okay, it's signed.

**Review by Ben-Louis — APPROVED** [review]

File: mmpose/models/heads/base_head.py [hunk 1]
```diff
@@ -63,9 +63,14 @@ def _get_in_channels(self) -> Union[int, List[int]]:
...
 
-    def _transform_inputs(self, feats: Tuple[Tensor]
-                          ) -> Union[Tensor, Tuple[Tensor]]:
+    def _transform_inputs(
+        self,
+        feats: Union[Tensor, Sequence[Tensor]],
+    ) -> Union[Tensor, Tuple[Tensor]]:
         """Transform multi scale features into the network input."""
+        if not isinstance(feats, Sequence):
+            return feats
```
Comment (ly015) [inline mmpose/models/heads/base_head.py:72]: I think it's better to warn the users that inputting a bare tensor doesn't meet our convention and the `input_transform` argument will be ignored.

Comment (Ben-Louis) [reply mmpose/models/heads/base_head.py:76]: OK, the warning has been added

**Pull request merged**